internals/overlord/logstate/manager.go

```diff
@@ -44,5 +100,10 @@ func (m *LogManager) Ensure() error {
 
 // Stop implements overlord.StateStopper and stops all log forwarding.
 func (m *LogManager) Stop() {
-	// TODO: implement
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	for _, gatherer := range m.gatherers {
+		gatherer.stop()
```

Comment:
Here we should wait for gatherers to finish and have their final chance to flush their buffers.